container/rules/aggregate_debian_packages.bzl

```diff
@@ -0,0 +1,98 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@io_bazel_rules_docker//container:container.bzl", _container = "container")
+load("@base_images_docker//package_managers:download_pkgs.bzl", _download = "download")
+load("@base_images_docker//package_managers:apt_key.bzl", _key = "key")
+
+debian_pkgs_attrs = _container.image.attrs + _key.attrs + _download.attrs + {
+    # Redeclare following attributes as non-mandatory.
+    "image_tar": attr.label(
+        allow_files = True,
+        single_file = True,
+    ),
+    "image": attr.label(
+        allow_files = True,
+        single_file = True,
+    ),
+    "packages": attr.string_list(),
+    "keys": attr.label_list(
+        allow_files = True,
+    ),
+}
+
+def _aggregate_debian_packages_impl(ctx):
+    """Implementation for the aggregate_debian_packages rule.
+    aggregate_debian_packages rule composes all attrs from itself and language_tool_layer(s),
+    and generates a tarball of debian packages using download_pkgs rule.
+    Args:
+      ctx: ctx for list of language_tool_layer(s)
+    """
+
+    packages = []
+    additional_repos = []
+    keys = []
+
+    for layer in ctx.attr.language_layers:
+        packages.extend(layer.packages)
+        additional_repos.extend(layer.additional_repos)
+        keys.extend(layer.keys)
+    packages.extend(ctx.attr.packages)
+    additional_repos.extend(ctx.attr.additional_repos)
+    keys.extend(ctx.files.keys)
+
+    packages = depset(packages).to_list()
+    additional_repos = depset(additional_repos).to_list()
+    keys = depset(keys).to_list()
+
+    download_base = ctx.files.base[0]
+
+    # Create an intermediate image with additional gpg keys used to download packages.
```

Comment:
This is all duplicating code in https://github.com/bazelbuild/bazel-toolchains/blob/master/container/rules/docker_toolchains.bzl#L97
remove it from there and leave this logic only here.